### PR TITLE
Add section on using scala3 union types

### PR DESCRIPTION
**Pull request by eloots**
Draft for doc update regarding the utilisation of Scala 3 Union types to eliminate message adapters & wrappers.

Would certainly appreciate feedback from @pjfanning and @He-Pin

### CODE REVIEW

**Comment by pjfanning** [comment]
@eloots Looks like this is on the right track. I can have a closer look in a day or 2.

Would you have interest in creating a PR to backport your original PR (with the example) to the `1.0.x` branch. We can also backport this PR when it is ready - meaning that the 1.0.x docs can be updated to include this.

File: actor-typed-tests/src/test/scala-3/docs/org/apache/pekko/typed/InteractionPatterns3Spec.scala [hunk 1]
```diff
@@ -138,19 +138,20 @@ class InteractionPatterns3Spec extends ScalaTestWithActorTestKit with AnyWordSpe
...
-        def apply(backend: ActorRef[Backend.Request]): Behavior[Command] =
+        def apply(backend: ActorRef[Backend.Request]): Behavior[Command] = // (2)
           Behaviors.setup[CommandAndResponse] { context =>
 
             def active(inProgress: Map[Int, ActorRef[URI]], count: Int): Behavior[CommandAndResponse] = {
               Behaviors.receiveMessage[CommandAndResponse] {
                 case Translate(site, replyTo) =>
                   val taskId = count + 1
-                  backend ! Backend.StartTranslationJob(taskId, site, context.self)
+                  val x: ActorRef[CommandAndResponse] = context.self
```
Comment (He-Pin) [inline actor-typed-tests/src/test/scala-3/docs/org/apache/pekko/typed/InteractionPatterns3Spec.scala:150]: `val x` is not used.

File: docs/src/main/paradox/typed/index.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ project.description: Using Apache Pekko to build reliable multi-core application
 * [actors](actors.md)
 * [actor-lifecycle](actor-lifecycle.md)
 * [interaction patterns](interaction-patterns.md)
+* [handling responses in Scala 3](handling-actor-responses-scala3.md)
```
Comment (He-Pin) [inline docs/src/main/paradox/typed/index.md:13]: `handling-actor-responses-with-scala3.md`

File: docs/src/main/paradox/typed/interaction-patterns.md [hunk 3]
```diff
@@ -112,7 +113,12 @@ Java
 
 ## Adapted Response
 
-Most often the sending actor does not, and should not, support receiving the response messages of another actor. In such cases we need to provide an @apidoc[actor.typed.ActorRef] of the right type and adapt the response message to a type that the sending actor can handle.
+Most often the sending actor does not, and should not, support receiving the response messages of another actor.
+
+In such cases we need to provide an @apidoc[actor.typed.ActorRef] of the right type and adapt the response message
+to a type that the sending actor can handle. In the case of Scala, we need to make a distinction between Scala 2
+and Scala 3. In the latter case, we can actually get rid of the need to adapt the response message which vastly
+simplifies the handling of responses. The details can be found in the section @ref:[Handling actor responses in Scala 3](handling-actor-responses-scala3.md).
```
Comment (He-Pin) [inline docs/src/main/paradox/typed/interaction-patterns.md:121]: By leveraging the `Union type`?

File: docs/src/main/paradox/typed/index.md (hunk 2 above)
Comment (eloots) [reply docs/src/main/paradox/typed/index.md:13]: Good suggestion! Will change the text in the link too.

**Review by He-Pin — APPROVED** [review]
lgtm

File: docs/src/main/paradox/typed/interaction-patterns.md (hunk 3 above)
Comment (eloots) [reply docs/src/main/paradox/typed/interaction-patterns.md]: Good point!
Forgot the add this to my last PR update.
Will push another update.

**Comment by eloots** [comment]
@pjfanning @He-Pin 

> We can also backport this PR when it is ready - meaning that the 1.0.x docs can be updated to include this.

I just created #741 with the backport

**Pull request merged**